src/components/App.js

```diff
@@ -1,20 +1,39 @@
+/* eslint react/forbid-prop-types: 0 */
+
 import React from 'react'
+import { connect } from 'react-redux'
 
 import Footer from './Footer'
 import Glossary from './Glossary'
 import Navigation from './Navigation'
 
-const App = props => (
-  <div className='site'>
-    <div className='site-wrap'>
-      <Navigation />
-      <main className='container'>
-        { props.children }
-      </main>
+const App = props => {
+  const appState = props.appState
+  const dispatch = props.dispatch
+
+  return (
+    <div className='site'>
+      <div className='site-wrap'>
+        <Navigation />
+        <main className='container'>
+          {props.children}
+        </main>
+      </div>
+      <Glossary
+        dispatch={dispatch}
+        {...appState.glossary}
+      />
+      <Footer />
     </div>
-    <Glossary />
-    <Footer />
-  </div>
-)
+  )
+}
+
+App.propTypes = {
+  appState: React.PropTypes.object,
+  dispatch: React.PropTypes.func,
+}
+
+const mapStateToProps = state => ({ appState: state })
+const mapDispatchToProps = dispatch => ({ dispatch })
```

Comment:
i'm not sure we need this -- i think dispatch should flow into the props by default without it